### PR TITLE
Setup

### PR DESCRIPTION
**Pull request by MateoLostanlen**
This PR is not ready, I'm having trouble with setup

### CODE REVIEW

File: .coveragerc [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[run]
+source = pyronear
```
Comment (frgfm) [inline .coveragerc:2]: this needs to be changed to the actual package folder name

File: .github/workflows/doc-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+name: doc-deploy
```
Comment (frgfm) [inline .github/workflows/doc-deploy.yml:1]: Since there is no documentation build right now in the repo, you can remove this file ;)

File: .github/workflows/gh-page.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+name: GH-Pages Status
```
Comment (frgfm) [inline .github/workflows/gh-page.yml:1]: Same here: you can remove this for now

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install pyronear
+          pip install --user picamera
+          pip install -r docs/requirements.txt
+
+      - name: Build documentation
+        run: |
+          sphinx-build docs/source docs/build -a
```
Comment (frgfm) [inline .github/workflows/main.yml:123]: You can remove this for now since documentation cannot be built

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+          key: ${{ runner.os }}-install-${{ env.cache-name }}-${{ hashFiles('requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-install-${{ env.cache-name }}-
+            ${{ runner.os }}-install-
+            ${{ runner.os }}-
+      - name: Install package
+        run: |
+          python -m pip install --upgrade pip
+          pip install --user picamera
+          pip install pyronear
```
Comment (frgfm) [inline .github/workflows/main.yml:36]: Since you need a RaspberryPI environment and not  `ubuntu-latest` we will need to use a docker I guess. So for now, I would recommend removing the job

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+        run: |
+          coverage run -m unittest discover test/
+          coverage xml
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v1.0.6
+        with:
+          file: ./coverage.xml
+          flags: unittests
+          fail_ci_if_error: true
```
Comment (frgfm) [inline .github/workflows/main.yml:76]: Same here: we'll add unittests once the docker is set up, you can remove this

File: .vscode/settings.json [hunk 7]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (frgfm) [inline .vscode/settings.json:1]: This should be removed along with .pyc files and pyronearEngine/raspberryPi/model/model_resnet18.txt and `test/.coverage`. Once removed, you might want to add a gitignore file similar to this one to avoid this later on (https://github.com/pyronear/PyroNear/blob/master/.gitignore) ;)

File: test/coverage.xml [hunk 8]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" ?>
```
Comment (frgfm) [inline test/coverage.xml:1]: Should be removed as well

File: test/test_utils.py [hunk 9]
```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
```
Comment (frgfm) [inline test/test_utils.py:1]: This cannot be run as is, only works on Raspberry PI, you can uncomment the appropriate sections, and we'll setup a docker to be able to run this on another PR

File: pyronearEngine/version.py [hunk 10]
```diff
@@ -0,0 +1 @@
+__version__ = '0.2.0a0'
```
Comment (frgfm) [inline pyronearEngine/version.py:1]: I think you might want to start with a `0.1.0.a0` rather than skipping to `0.2.0..` ;)

File: requirements.txt [hunk 11]
```diff
@@ -0,0 +1,8 @@
+opencv-python>=3.4.5.20
+pafy>=0.5.5
+pandas>=0.25.2
+PyYAML>=5.1.2
+torch>=1.2.0
+torchvision>=0.4.0
+tqdm>=4.20.0
+requests>=2.20.0
```
Comment (frgfm) [inline requirements.txt:8]: Only put minimal requirements, most of these dependencies are not used in your code

File: setup.py [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+    requirements = f.read().splitlines()
+
+setup(
+    # Metadata
+    name=package_name,
+    version=version,
+    author='PyroNear Contributors',
+    author_email='pyronear.d4g@gmail.com',
+    maintainer='Pyronear',
+    description='Datasets and models for wildfire detection in PyTorch',
```
Comment (frgfm) [inline setup.py:49]: This here should be edited

File: pyronearEngine/raspberryPi/pyronearEngine.py [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+        else:
+            sendAlert()
+            return "FIRE !!!"
+
+
+def sendAlert():
+    port = 587  # For starttls
+    smtp_server = "smtp.gmail.com"
+    sender_email = "test.pyronear@gmail.com"  # Enter your address
+    receiver_email = "mateo.lostanlen@gmail.com"  # Enter receiver address
```
Comment (frgfm) [inline pyronearEngine/raspberryPi/pyronearEngine.py:91]: I would highly suggest to remove your personal address from a PR that will be merged into a public commit :sweat_smile:

File: pyronearEngine/raspberryPi/pyronearEngine.py [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+        self.imgsFolder = imgsFolder
+
+        # Model definition
+        self.model = torchvision.models.resnet18(pretrained=True)
+
+        # Change fc
+        in_features = getattr(self.model, 'fc').in_features
+        setattr(self.model, 'fc', nn.Linear(in_features, 2))
+
+        self.model.load_state_dict(torch.load("model/model_resnet18.txt", map_location=torch.device('cpu')))
```
Comment (frgfm) [inline pyronearEngine/raspberryPi/pyronearEngine.py:43]: Would be better to store the model on the main repo attachments, but we cannot add the binary file to the PR

**Review by frgfm — CHANGES_REQUESTED** [review]
Hey there, thanks for the PR!
I put comments where I thought we might need to change something. Let me know what you think, cheers!

File: setup.py [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+    requirements = f.read().splitlines()
+
+setup(
+    # Metadata
+    name=package_name,
+    version=version,
+    author='PyroNear Contributors',
+    author_email='pyronear.d4g@gmail.com',
+    maintainer='Pyronear',
+    description='Pyronear Engine is a repository that aims to deploy pyronear',
```
Comment (frgfm) [inline setup.py:49]: small typo here: "that aims at deploying" ;)

File: .coveragerc (hunk 1 above)
Comment (MateoLostanlen) [reply .coveragerc]: Done

File: .github/workflows/doc-deploy.yml (hunk 2 above)
Comment (MateoLostanlen) [reply .github/workflows/doc-deploy.yml]: Done

File: .github/workflows/gh-page.yml (hunk 3 above)
Comment (MateoLostanlen) [reply .github/workflows/gh-page.yml]: Done

File: .github/workflows/main.yml (hunk 4 above)
Comment (MateoLostanlen) [reply .github/workflows/main.yml]: Done

File: .github/workflows/main.yml (hunk 5 above)
Comment (MateoLostanlen) [reply .github/workflows/main.yml]: Ok I will check that

File: requirements.txt (hunk 11 above)
Comment (MateoLostanlen) [reply requirements.txt]: Done

File: setup.py (hunk 12 above)
Comment (MateoLostanlen) [reply setup.py]: Done

File: pyronearEngine/raspberryPi/pyronearEngine.py (hunk 13 above)
Comment (MateoLostanlen) [reply pyronearEngine/raspberryPi/pyronearEngine.py]: Oups :)

**Review by frgfm — COMMENTED** [review]
Still a few things to get the PR ready, but we're getting there! There are still some files to remove: .coverage, .vscode/settings.json, coverage.xml, pyronearEngine/__pycache__, pyronearEngine/raspberryPi/__pycache__, test/.coverage, test/__pycache__, test/coverage.xml

File: setup.py (hunk 15 above)
Comment (MateoLostanlen) [reply setup.py:49]: Thanks :)

File: pyronearEngine/raspberryPi/pyronearEngine.py (hunk 14 above)
Comment (MateoLostanlen) [reply pyronearEngine/raspberryPi/pyronearEngine.py]: Ok done

**Comment by MateoLostanlen** [comment]
Hi @frgfm , thanks for your review. I cleaned the PR has you asked. I still have an issue with the CI setup:
``` 
x> No CI provider detected.
    Testing inside Docker? http://docs.codecov.io/docs/testing-with-docker
    Testing with Tox? https://docs.codecov.io/docs/python#section-testing-with-tox
    project root: .
    Yaml not found, that's ok! Learn more at http://docs.codecov.io/docs/codecov-yaml
    -> Found 1 reports
==> Detecting git/mercurial file structure
==> Reading reports
    + ./coverage.xml bytes=986
==> Appending adjustments
    https://docs.codecov.io/docs/fixing-reports
    -> No adjustments found
==> Gzipping contents
==> Uploading reports
    url: https://codecov.io
    query: branch=master&commit=6aebc232fa4e9dadf98c2b4b10102e407f8f9f06&build=&build_url=&name=&tag=&slug=MateoLostanlen%2FPyronearEngine&service=&flags=unittests&pr=&job=&cmd_args=f,n,F,Z
->  Pinging Codecov
https://codecov.io/upload/v4?package=bash-20200910-5a68dc2&token=secret&branch=master&commit=6aebc232fa4e9dadf98c2b4b10102e407f8f9f06&build=&build_url=&name=&tag=&slug=MateoLostanlen%2FPyronearEngine&service=&flags=unittests&pr=&job=&cmd_args=f,n,F,Z
HTTP 400
Please provide the repository token to upload reports via `-t :repository-token`
##[error]Codecov failed with the following error: The process 'bash' failed with exit code 1
````

Do you know how to solve this ? It seems I have a codecov error access, however from what I understand there is no need for special setup on a public repo

**Comment by frgfm** [comment]
Hey there :wave: 
Yes indeed, I actually talked with them about that for another project. In the main workflow, replace `uses: codecov/codecov-action@v1.0.6` with `uses: codecov/codecov-action@v1` which will use the latest version available (which includes the fix apparently). Let me know if you still have issues!

**Comment by MateoLostanlen** [comment]
@frgfm Perfect, it works :)

File: pyronearEngine/raspberryPi/pyronearEngine.py [hunk 16]
```diff
@@ -0,0 +1,110 @@
...
+        self.imgsFolder = imgsFolder
+
+        # Model definition
+        self.model = torchvision.models.resnet18(pretrained=False)
+
+        # Change fc
+        in_features = getattr(self.model, 'fc').in_features
+        setattr(self.model, 'fc', nn.Linear(in_features, 2))
+
+        self.model.load_state_dict(torch.load("model/pyronear.pth", map_location=torch.device('cpu')))
```
Comment (frgfm) [inline pyronearEngine/raspberryPi/pyronearEngine.py:43]: I think it would be better to pass the checkpoint path to the constructor, so that the user can easily specify her/his own!

**Review by frgfm — COMMENTED** [review]
Apart from the checkpoint path that would benefit from being dynamic, I think we're good to go!
Perhaps check my last comment to see what you think and then we'll merge the PR :+1:
